### PR TITLE
rustdoc: change the +/- icons

### PR DESCRIPTION
**Pull request by rol1510**
The `[+]` and `[-]` icons to expand/collapse parts of the documentation are really difficult to tell apart. This tries to replace them with something simpler.

Old
image

New
image
image
image

Demo: https://host-7e0a2.web.app/sample-2/std/iter/trait.Iterator.html

Also discussed here: #59851

### CODE REVIEW

File: src/librustdoc/html/static/css/rustdoc.css [hunk 1]
```diff
@@ -1448,9 +1448,9 @@ details.toggle > summary.hideme > span {
 
 details.toggle > summary::before {
 	/* toggle plus */
-	background: url('data:image/svg+xml,<svg width="17" height="17" \
-shape-rendering="crispEdges" stroke="black" fill="none" xmlns="http://www.w3.org/2000/svg"><path \
-d="M5 2.5H2.5v12H5m7-12h2.5v12H12M5 8.5h7M8.5 12V8.625v0V5"/></svg>') no-repeat top left;
+	background: url('data:image/svg+xml,<svg width="17" height="17" fill="gray" \
+	stroke="gray"	xmlns="http://www.w3.org/2000/svg"><path d="M13 8.5L4 14L4 3L13 8.5Z"/> \
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/css/rustdoc.css:1452]: Weird spacing after "gray".

File: src/librustdoc/html/static/css/rustdoc.css [hunk 2]
```diff
@@ -1529,9 +1529,9 @@ details.toggle[open] > summary.hideme > span {
 
 details.toggle[open] > summary::before {
 	/* toggle minus */
-	background: url('data:image/svg+xml,<svg width="17" height="17" \
-shape-rendering="crispEdges" stroke="black" fill="none" xmlns="http://www.w3.org/2000/svg"><path \
-d="M5 2.5H2.5v12H5m7-12h2.5v12H12M5 8.5h7"/></svg>') no-repeat top left;
+	background: url('data:image/svg+xml,<svg width="17" height="17" fill="gray" \
+	stroke="gray"	xmlns="http://www.w3.org/2000/svg"><path d="M8.5 13L3 4L14 4L8.5 13Z"/> \
```
Comment (GuillaumeGomez) [inline src/librustdoc/html/static/css/rustdoc.css:1533]: Weird spacing after "gray".

**Comment by GuillaumeGomez** [comment]
I also like the new icon and I agree that we need an FCP for it.

Can you provide screenshots for the other themes as well please?

**Comment by tgross35** [comment]
(nit) It looks like for the the methods and description dropdown, the arrow appears to be slightly below the line of text. Compared to function & type items where it's centered

image

image

(also nit) A slight size reduction to about 90% of line height might look good, just to reduce its prominence compared to the text it's associated with

**Comment by rol1510** [comment]
The toggle-all-docs button now also has a new icon. Also the other icons are now little smaller and they are aligned properly.

**Comment by GuillaumeGomez** [comment]
I'm not a big fan of the new "toggle all" button. At least the previous was somehow explicit about what it was doing. The new one is more beautiful but more difficult to decipher (imo).

**Comment by jsha** [comment]
Re: the toggle button, I think the most explicit thing would be to replace the icon with a word - `summary` in the default state, or `expand` after everything has been collapsed.

**Comment by rol1510** [comment]
I like @jsha's Idea. Looks great and is very explicit.

**Comment by GuillaumeGomez** [comment]
I like it too. Should we use the same color as for the `source` link right beside it?

**Comment by rol1510** [comment]
I feel like the gray looks a little better. Also, the color of `source` is the link color, which could be confusing.

**Comment by GuillaumeGomez** [comment]
But grey is associated with "disabled item". So I don't think it's a good idea to keep as is.

**Comment by rol1510** [comment]
Good Point. We could set it to the main color

**Comment by rol1510** [comment]
A border around it might also work

**Comment by rol1510** [comment]
maybe something like that?

Frame 8

**Comment by tgross35** [comment]
If this has been decided against, it probably makes sense to post an update to #59851. But maybe Just making the **+** and **-** bold would be enough to address the original listed "difficult to tell apart" concern?

**Pull request closed without merging**